WordPress/src/test/java/org/wordpress/android/viewmodel/reader/ReaderPostListViewModelTest.kt

```diff
@@ -351,6 +360,90 @@ class ReaderPostListViewModelTest {
         assertThat(viewModel.currentSubFilter.value).isEqualTo(null)
     }
 
+    @Test
+    fun `when user has no site the no site flow is triggered`() {
+        val siteId = 1
+        val site = SiteModel()
+        val post = ReaderPost()
+        val visibleSites = listOf<SiteModel>() // No sites
+
+        whenever(appPrefsWrapper.getSelectedSite()).thenReturn(siteId)
+        whenever(siteStore.getSiteByLocalId(siteId)).thenReturn(site)
+        whenever(siteStore.visibleSites).thenReturn(visibleSites)
+
+        viewModel.onReblogButtonClicked(post)
+
+        val state = viewModel.reblogAction.value?.peekContent()
+        assertThat(state).isEqualTo(NoSite)
+    }
+
+    @Test
+    fun `when user has only one site the post editor is triggered`() {
+        val siteId = 1
+        val site = SiteModel()
+        val post = ReaderPost()
+        val visibleSites = listOf(site) // One site
+
+        whenever(appPrefsWrapper.getSelectedSite()).thenReturn(siteId)
+        whenever(siteStore.getSiteByLocalId(siteId)).thenReturn(site)
+        whenever(siteStore.visibleSites).thenReturn(visibleSites)
+
+        viewModel.onReblogButtonClicked(post)
+
+        val state = viewModel.reblogAction.value?.peekContent()
+        assert(state is PostEditor)
+        assertThat(state?.site).isEqualTo(site)
+        assertThat(state?.post).isEqualTo(post)
+    }
+
+    @Test
+    fun `when user has more than one sites the site picker is triggered`() {
+        val siteId = 1
+        val site = SiteModel()
+        val post = ReaderPost()
+        val visibleSites = listOf(site, site) // More sites
+
+        whenever(appPrefsWrapper.getSelectedSite()).thenReturn(siteId)
+        whenever(siteStore.getSiteByLocalId(siteId)).thenReturn(site)
+        whenever(siteStore.visibleSites).thenReturn(visibleSites)
+
+        viewModel.onReblogButtonClicked(post)
+
+        val state = viewModel.reblogAction.value?.peekContent()
+        assert(state is SitePicker)
+        assertThat(state?.site).isEqualTo(site)
+        assertThat(state?.post).isEqualTo(post)
+    }
+
+    @Test
+    fun `when user has more than one sites and selects the site to reblog the post editor is triggered`() {
+        val siteId = 1
+        val site = SiteModel()
+        val post = ReaderPost()
+        val visibleSites = listOf(site, site) // More sites
+
+        whenever(appPrefsWrapper.getSelectedSite()).thenReturn(siteId)
+        whenever(siteStore.getSiteByLocalId(siteId)).thenReturn(site)
+        whenever(siteStore.visibleSites).thenReturn(visibleSites)
+
+        viewModel.onReblogButtonClicked(post)
+        viewModel.selectedSiteToReblog(siteId)
+
+        val state = viewModel.reblogAction.value?.peekContent()
+        assert(state is PostEditor)
+        assertThat(state?.site).isEqualTo(site)
```

Comment:
This can be a personal preference 😊; in general I like the assertThat readability more than the pure assert function. Wdyt in using them also for things like `assert(state is PostEditor)`? Something like `assertThat(state).isInstanceOf(PostEditor::class.java)`?

Reply:
Tbh I don't like the verbosity of the `::class.java` suffix.
It would be more concise with the rest of the tests though.
Fixed cf3017354faf1278810179b5955dcbf3b01d6a9e